atlasdb-workload-server-api/src/main/java/com/palantir/atlasdb/workload/transaction/witnessed/WitnessedRowColumnRangeReadTransactionAction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.transaction.witnessed;
+
+import com.palantir.atlasdb.workload.transaction.RowColumnRangeReadTransactionAction;
+import java.util.Map;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface WitnessedRowColumnRangeReadTransactionAction extends WitnessedTransactionAction {
+    RowColumnRangeReadTransactionAction originalQuery();
+
+    Map<Integer, Integer> columnsAndValues();
```

Comment:
unused in this PR: I change this to SortedMap in Part 3, can backport if you wish though I don't think this changes much of the review (and there are no users yet)

Reply:
ugh, actually these types are both wrong. realistically we want a List of some value type, because order matters.